src/btree/bt_page.c

```diff
@@ -570,7 +572,9 @@ __inmem_row_leaf(WT_SESSION_IMPL *session, WT_PAGE *page, bool check_unstable)
              * Simple keys without compression (not Huffman encoded or prefix compressed), can be
              * directly referenced on the page to avoid repeatedly unpacking their cells.
              */
-            if (!btree->huffman_key && unpack.prefix == 0)
+            if (!btree->huffman_key && unpack.prefix == 0 && unpack.stop_txn == WT_TXN_MAX &&
+              unpack.stop_ts == WT_TS_MAX && txn_global->has_oldest_timestamp &&
+              unpack.start_ts < txn_global->oldest_timestamp)
```

Comment:
I'm confused, keys don't have timestamps, do they?

Reply:
417ce1c

You're right. This check is redundant for key cells.